modules/_internal/runner/main.tf

```
@@ -95,32 +82,17 @@ data "aws_iam_policy_document" "codebuild" {
 }
 
 resource "aws_iam_role" "codebuild" {
-  name_prefix        = "flow-ci-codebuild-service-role-"
-  description        = "${local.name_slug}-codebuild-service-role -- Managed by Terraform"
-  assume_role_policy = data.aws_iam_policy_document.codebuild_assume_role.json
-}
-
-resource "aws_iam_role_policy" "codebuild" {
-  name   = "${local.name_slug}-codebuild"
-  role   = aws_iam_role.codebuild.id
-  policy = data.aws_iam_policy_document.codebuild.json
-}
-
-data "aws_iam_policy" "codebuild" {
-  count = length(var.policy_arns)
-
-  arn = data.template_file.policy_arns[count.index].rendered
-}
-
-resource "aws_iam_role_policy_attachment" "codebuild" {
-  count = length(var.policy_arns)
-
-  role       = aws_iam_role.codebuild.id
-  policy_arn = element(data.aws_iam_policy.codebuild.*.arn, count.index)
+  name_prefix         = "flow-ci-codebuild-service-role-"
+  description         = "${local.name_slug}-codebuild-service-role -- Managed by Terraform"
+  assume_role_policy  = data.aws_iam_policy_document.codebuild_assume_role.json
+  managed_policy_arns = var.policy_arns
```

Keeping the template_file reference, this would be something like:

```suggestion
  managed_policy_arns = data.template_file.policy_arns[*].rendered
```